veracruz-server-sgx/fuzz/fuzz_targets/provision_data.rs

```diff
@@ -0,0 +1,314 @@
+    client_config.set_single_client_cert(client_cert_vec, client_priv_key);
+    client_config
+	.root_store
+	.add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
+
+    client_config.pinned_cert_hashes.push(cert_hash);
+
+    let enclave_name = veracruz_server
+	.get_enclave_name()
+	.expect("Error obtaining Enclave name");
+    let dns_name_ret = webpki::DNSNameRef::try_from_ascii_str(enclave_name.as_str());
+    assert!(dns_name_ret.is_ok());
+    let dns_name = dns_name_ret.expect(&format!(
+	"Error obtaining DNS name reference from enclave {}",
+	enclave_name
+    ));
+    rustls::ClientSession::new(&std::sync::Arc::new(client_config), dns_name)
+}
+
+
+
+fn client_tls_send(
+    tx: &std::sync::mpsc::Sender<std::vec::Vec<u8>>,
+    rx: &std::sync::mpsc::Receiver<std::vec::Vec<u8>>,
+    session: &mut dyn rustls::Session,
+    send_data: &[u8],
+) -> Result<Vec<u8>, String> {
+    session
+        .write_all(&send_data)
+        .map_err(|_| format!("Error writing session data: {:?}", send_data))?;
+
+    let mut output: std::vec::Vec<u8> = std::vec::Vec::new();
+
+    session
+        .write_tls(&mut output)
+        .map_err(|_| format!("Error writing TLS data: {:?}", output))?;
+
+    let cloned = &output.clone();
+
+    tx.send(output)
+        .map_err(|_| format!("Error sending output: {:?}", cloned))?;
+
+    loop {
+        let received = rx.try_recv();
+        let cloned = &received.clone();
+
+        if received.is_ok() && (!session.is_handshaking() || session.wants_read()) {
+            let received =
+                received.map_err(|_| format!("Error interpreting received data: {:?}", cloned))?;
+
+            let mut slice = &received[..];
+            session
+                .read_tls(&mut slice)
+                .map_err(|_| format!("Error reading TLS data: {:?}", slice))?;
+            session
+                .process_new_packets()
+                .map_err(|_| "Error processing new packets".to_string())?;
+
+            let mut received_buffer: std::vec::Vec<u8> = std::vec::Vec::new();
+
+            match session.read_to_end(&mut received_buffer) {
+                Err(reason) => {
+                    return Err(format!(
+                        "client_tls_send: session.read_to_end failed with {}",
+                        reason
+                    ))
+                }
+                Ok(num_bytes) => {
+                    if num_bytes > 0 {
+                        return Ok(received_buffer);
+                    }
+                }
+            }
+        } else if session.wants_write() {
+            let mut output: std::vec::Vec<u8> = std::vec::Vec::new();
+            session
+                .write_tls(&mut output)
+                .map_err(|_| format!("Error writing TLS data: {:?}", output))?;
+            let res = tx.send(output);
+
+            match res {
+                Ok(_) => (),
+                Err(err) => return Err(format!("Error in tx.send() {:?}", err)),
```

Comment:
`map_err` in `Result` type and `ok_or` in `Option` type, together with the `?` operator,  can be useful in term of propagating Error, e.g.
```
let res = tx.send(output).map_err(|err|format!("Error in tx.send() {:?}", err))?;
```
In fact here if you never want to use the `Ok(_)` case, you could try
```
tx.send(output).map_err(|err| format!("Error in tx.send() {:?}", err))?;
```
Just to say, there is no problem of use `match`. However, these two functions can help you to write less code, especially less indentation for readability.